apps/src/aiTutor/redux/aiTutorRedux.ts

```diff
@@ -64,6 +69,24 @@ export const askAITutor = createAsyncThunk(
     );
 
     thunkAPI.dispatch(addAIResponse(chatApiResponse?.content));
+    const prompt = ChatContext.systemPrompt + ChatContext.studentCode;
+    const type =
+      ChatContext.systemPrompt === validationSystemPrompt
+        ? aiTutorTypes.VALIDATION
+        : aiTutorTypes.COMPILATION;
+
+    const interactionData = {
+      userId: 123456789,
+      levelId: ChatContext.levelId,
+      scriptId: 123,
+      type: type,
+      prompt: prompt,
+      status: chatApiResponse?.status,
+      aiResponse: chatApiResponse?.content,
+    };
+
+    console.log('interactionData', interactionData);
+    savePromptAndResponse(interactionData);
```

Comment:
@kakiha11 If you're up for it, you can start fleshing out the `interactionData` for the General Chat thunk (below). It should have a lot of similarities to this one. This general shape of data is what we'll set up for tests too so getting familiar with all the pieces we need will be helpful.

If you're feeling ambitious, you could start thinking about how to get `userId` and `scriptId` piped through, which you could do by tracing the path for how we're getting access to `levelId` from within the AI Tutor components.